plugins/stack-overflow/src/api/StackOverflowClient.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fetch from 'cross-fetch';
+import qs from 'qs';
+import { StackOverflowApi } from './StackOverflowApi';
+import {
+  StackOverflowQuestion,
+  StackOverflowQuestionsRequestParams,
+} from '../types';
+import { Config } from '@backstage/config';
+
+export class StackOverflowClient implements StackOverflowApi {
+  private baseUrl: string;
+
+  constructor({ baseUrl }: { baseUrl: string }) {
```

Comment:
with the fromConfig method added we should be able to make the constructor `private`